### PR TITLE
Update build system for internal Slice files in C++

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR simplifies the build system for internal Slice files used by IceGrid and IceStorm. 

The header files are now generated next to the .cpp files.

### CODE REVIEW

File: cpp/src/IceStorm/DBTypes.ice [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 [["cpp:header-ext:h"]]
 
-#include <IceStorm/SubscriberRecord.ice>
-#include <IceStorm/LLURecord.ice>
+#include "SubscriberRecord.ice"
```
Comment (bernardnormier) [inline cpp/src/IceStorm/DBTypes.ice:9]: The generated code still uses `<...>` which we should probably fix.

File: cpp/src/IceStorm/Makefile.mk [hunk 2]
```diff
@@ -4,8 +4,6 @@
 
 $(project)_libraries            := IceStormService
 $(project)_programs             := icestormadmin icestormdb
-$(project)_sliceflags           := -Isrc --include-dir IceStorm
```
Comment (bernardnormier) [inline cpp/src/IceStorm/Makefile.mk:7]: Note that this PR does not add (or remove) any include directory from the C++ build.

File: cpp/src/IceStorm/msbuild/icestormadmin/icestormadmin.vcxproj [hunk 3]
```diff
@@ -84,9 +84,12 @@
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Label="IceBuilder">
     <SliceCompile>
-      <IncludeDirectories>$(IceSrcRootDir)\src\;%(IncludeDirectories)</IncludeDirectories>
-      <HeaderOutputDir>$(Platform)\$(Configuration)\IceStorm</HeaderOutputDir>
-      <BaseDirectoryForGeneratedInclude>IceStorm</BaseDirectoryForGeneratedInclude>
+      <IncludeDirectories>
```
Comment (bernardnormier) [inline cpp/src/IceStorm/msbuild/icestormadmin/icestormadmin.vcxproj:87]: I cleared these settings with the GUI. Not sure why it results in empty definitions.

Comment (pepone) [reply cpp/src/IceStorm/msbuild/icestormadmin/icestormadmin.vcxproj:87]: There is an option to set to inherit, which gets rid of it.

Comment (bernardnormier) [reply cpp/src/IceStorm/msbuild/icestormadmin/icestormadmin.vcxproj:87]: This works, but it's oddly not persistent. After you save, you can't tell the difference between inherit and blank.

Comment (bernardnormier) [reply cpp/src/IceStorm/msbuild/icestormadmin/icestormadmin.vcxproj]: Updated the project files.

The difference is once you've selected "Inherit from project defaults...", you don't see this option anymore.

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

**Pull request merged**